test/validator.test.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint-disable no-undef */
+"use strict";
+import { expect, default as chai } from "chai";
+import sinon from "sinon";
+import chaiAsPromised from "chai-as-promised";
+import { printResults, validateFile } from "../src/validator";
+import {
+  getHappySpec,
+  renderSpecAsFile,
+  getSingleInvalidFile
+} from "./testUtils";
+
+const PROFILE = "mercury";
+
+before(() => {
+  chai.use(chaiAsPromised);
+});
+
+describe("#printResults", () => {
+  let consoleSpy;
+
+  beforeEach(() => {
+    consoleSpy = sinon.spy(console, "log");
+  });
+
+  afterEach(() => {
+    consoleSpy.restore();
+  });
+
+  it("doesn't console log when no results passed", async () => {
+    await printResults(null);
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
+    return expect(consoleSpy.callCount).to.equal(0);
+  });
+
+  it("outputs results in first branch when results conform and warnings are false (default)", async () => {
+    const doc = getHappySpec();
+    const result = await validateFile(renderSpecAsFile(doc), PROFILE);
+
+    await printResults(result);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
+    return expect(consoleSpy.getCall(0).args[0]).to.have.string(
+      "Conforms? true"
+    );
+  });
+
+  it("outputs results in second branch when results do not conform and warnings are false (default)", async () => {
+    const doc = getSingleInvalidFile();
+    const result = await validateFile(renderSpecAsFile(doc), PROFILE);
+
+    await printResults(result);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
+    return expect(consoleSpy.getCall(0).args[0]).to.have.string(
```

Comment:
Same as above.  To elaborate the `console.log(results.toString());` will produce the message of `Conforms? false|true` just like our print does.

Reply:
No, in this case because warnings are false the only way `Conforms? false` can appear in the console log is if the results have conforms false, meaning we are in the second branch

Reply:
Unless our if logic is wrong.  All I am saying is that regardless of the conditional this line as is CAN appear in either branch.  

If we make the assumption that our conditional logic is correct then why test the branches?

If someone makes a change in the future to the conditional logic that breaks it this should cause a failure and I'm not sure it will here. 

I won't block it on these grounds but would love to see this updated.

Reply:
Updated to be more specific